spec/when_committed_spec.rb

```diff
@@ -105,6 +124,60 @@
       end
       Backgrounder.should have(1).job
     end
+
+    describe "nested transactions" do
+      it "runs the provided block once, after the outer transaction is committed" do
+        Widget.transaction do
+          model.action_that_needs_follow_up_after_commit
+          Widget.transaction(requires_new: true) do
+            model.another_action_with_follow_up
+          end
+          Backgrounder.jobs.should == []
+        end
+        Backgrounder.jobs.should == [:important_work, :more_work]
+      end
+
+      it "does not run the block if it is defined in a nested transaction that is rolled back" do
+        Widget.transaction do
+          model.action_that_needs_follow_up_after_commit
+          Widget.transaction(requires_new: true) do
+            model.another_action_with_follow_up
+            raise ActiveRecord::Rollback
+          end
+          Backgrounder.jobs.should == []
+        end
+        Backgrounder.jobs.should == [:important_work]
+      end
```

Comment:
I added these specs to make sure it behaved as expected with nested transactions. This one specifically shows that the commit block doesn't run for a nested transaction that was rolled back.